manifests/mount.pp

```diff
@@ -37,52 +37,43 @@
 # Copyright 2014 CoverMyMeds, unless otherwise noted
 #
 define gluster::mount (
-  $volume         = undef,
-  $log_level      = undef,
-  $log_file       = undef,
-  $transport      = undef,
-  $direct_io_mode = undef,
-  $readdirp       = undef,
-  $atboot         = yes,
-  $options        = 'defaults',
-  $dump           = 0,
-  $pass           = 0,
-  $ensure         = mounted, # defined, present, unmounted, absent, mounted.
+  String $volume                                                        = undef,
+  Optional[String] $log_level                                           = undef,
+  Optional[String] $log_file                                            = undef,
+  Optional[String] $transport                                           = undef,
+  Optional[String] $direct_io_mode                                      = undef,
+  Optional[Boolean] $readdirp                                           = undef,
+  $atboot                                                               = yes,
+  String $options                                                       = 'defaults',
+  Integer $dump                                                         = 0,
+  Integer $pass                                                         = 0,
+  Enum['defined', 'present', 'unmounted', 'absent', 'mounted'] $ensure  = 'mounted',
 ) {
 
   if ! $volume or empty($volume) {
     fail('Volume parameter is mandatory for gluster::mount')
   }
 
   if $log_level {
-    validate_string($log_level)
```

Comment:
I would expect log_level to be in the list ['Emergency', 'Alert',  'Error', 'Warning', 'Notice', 'Info',  'Debug' ], Should that be validated that it's correct?